### PR TITLE
chore(fuzz): differential fuzzing for validation

### PR DESCRIPTION
**Pull request by lrlna**
This works of a branch in `routet-bridge` that calls js validation. This should be a more accurate representation of what happens in the router, making this more of Gala vs. Golden Delicious apple comparison.

### CODE REVIEW

File: fuzz/fuzz_targets/validation.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    compiler.add_document(&ts, "ts.graphql");
+    let rust_diagnostics = compiler.validate();
+
+    debug!("======= OPERATION =======");
+    debug!("{}", op);
+    debug!("========================");
+    debug!("{}", ts);
+    debug!("========================");
+    let js_diagnostics =
+        router_bridge::validate::validate(&ts, &op).expect("js validation should succeed");
```
Comment (goto-bus-stop) [inline fuzz/fuzz_targets/validation.rs:30]:
```suggestion
        router_bridge::validate::validate(&ts, &op).expect("bridge should return validation result");
```
would emphasise here that it's not that JS validation did not pass, but that something else went wrong with Deno or somewhere, not the actual GraphQL validation

File: fuzz/fuzz_targets/validation.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    let (op, ts) = match generate_valid_operation(data) {
+        Ok(d) => (d.0, d.1),
+        Err(_err) => {
+            return;
+        }
+    };
+
+    let mut compiler = ApolloCompiler::new();
+    compiler.add_executable(&op, "op.graphql");
+    compiler.add_document(&ts, "ts.graphql");
```
Comment (goto-bus-stop) [inline fuzz/fuzz_targets/validation.rs:21]:
```suggestion
    compiler.add_type_system(&ts, "ts.graphql");
```
just to match actual usage, the behaviour of `document` is a bit different

File: Cargo.toml [hunk 3]
```diff
@@ -8,3 +8,11 @@ members = [
     "crates/apollo-smith",
     "fuzz",
 ]
+
+default-members = [
```
Comment (lrlna) [inline Cargo.toml:12]: temporarily removing fuzz from workspace members for CI. not a good idea to merge it like this though.

Comment (lrlna) [reply Cargo.toml:12]: (router-bridge is extremely fussy and is refusing to build in CI due to too many failures)

**Comment by lrlna** [comment]
Closing this. Moved any deno-related validation to [its own repo](https://github.com/apollographql/apollo-rs-fuzz) so we don't have to build it as part of apollo-rs.

**Pull request closed without merging**